crates/orderbook/src/orderbook.rs

```diff
@@ -403,6 +406,66 @@ impl Orderbook {
             .await
             .context("get_user_orders error")
     }
+
+    pub async fn get_order_status(&self, uid: &OrderUid) -> Result<Option<dto::order::Status>> {
+        match self.database.latest_order_event(uid).await? {
```

Comment:
nit: Here `?` forwards db connection errors as `OrderNotFound` to the caller of `get_order_status`, not sure if desirable

Reply:
Not really, it returns a DB error which is then mapped to the internal error:
https://github.com/cowprotocol/services/blob/9a2745bed2cbcaa260d0cbe1e646ce88c42b417d/crates/orderbook/src/api/get_order_status.rs#L29-L32

Reply:
Ah yes, my bad